### PR TITLE
Unified naming of health checker in code and tests + Task creation refactoring

### PR DESCRIPTION
**Pull request by jwair**
- Unified naming of health checker in code and tests
- Updated oapi-codegen module

### CODE REVIEW

File: health/checker.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package health
 
 import "fmt"
 
-//go:generate mockery --name=Manager --filename=manager.go --output=../mocks/health
+//go:generate mockery --name=Checker --filename=checker.go --output=../mocks/health
```
Comment (jwair) [inline health/checker.go:5]: This is the main change in the PR, to unify the use of 'Checker' over 'Manager'

Comment (wilkermichael) [reply health/checker.go:5]: ah thanks for catching this, I did the refactor but then didn't catch these

File: e2e/condition_catalog_service_test.go [hunk 2]
```diff
@@ -369,10 +369,6 @@ func TestCondition_CatalogServices_SharedDependency_NoServices(t *testing.T) {
 		name    string
 		runMode string
 	}{
-		{
```
Comment (jwair) [inline e2e/condition_catalog_service_test.go:372]: Removed this invalid test case. Empty string for `run` is not a valid value.

File: api/task_lifecycle_client.go [hunk 3]
```diff
@@ -59,29 +59,12 @@ func (c *TaskLifecycleClient) Scheme() string {
...
-		return nil, err
-	}
-
-	resp, err := c.CreateTaskWithResponse(ctx, &oapigen.CreateTaskParams{Run: &run}, oapigen.CreateTaskJSONRequestBody(req))
-	if err != nil {
-		return nil, err
-	}
-
-	return resp, nil
+	run := oapigen.CreateTaskParamsRun(runOption)
```
Comment (jwair) [inline api/task_lifecycle_client.go:66]: Swagger validation happens as part of `c.CreateTaskWithResponse()` and that validates the run parameter is in the enum of supported options, no need for our additional validation logic.

File: api/task_lifecycle_client.go [hunk 4]
```diff
@@ -59,31 +58,6 @@ func (c *TaskLifecycleClient) Scheme() string {
 	return c.url.Scheme
 }
 
-// CreateTaskSimple takes a task request and run option and sends this information to the client. It then returns
-// a task response object and any errors to the caller.
-// TODO: remove this to conform to interface
-func (c *TaskLifecycleClient) CreateTask(ctx context.Context, runOption string, req TaskRequest) (*oapigen.CreateTaskResponse, error) {
-	var run oapigen.CreateTaskParamsRun
-	switch runOption {
```
Comment (jwair) [inline api/task_lifecycle_client.go:67]: Swagger validation happens in the generated client code, and that validates the run parameter is in the enum of supported options, no need for our additional validation logic.

File: e2e/condition_catalog_service_test.go (hunk 2 above)
Comment (wilkermichael) [reply e2e/condition_catalog_service_test.go:372]: actually in this case, we should verify that the nil case works. This test is verifying that the task creates without issue, and creating a task without a `run` parameter is a valid scenario from the API

**Review by wilkermichael — APPROVED** [review]
Comment about the nil `run` parameter case, otherwise everything else looks good

File: e2e/condition_catalog_service_test.go (hunk 2 above)
Comment (jwair) [reply e2e/condition_catalog_service_test.go:372]: Good catch.
Added back a test case for creating a task without run mode: https://github.com/hashicorp/consul-terraform-sync/pull/826/commits/e8737b9bfd9fe4ec549f9e9dbe9b91a8a21d744d

**Pull request merged**